src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -1032,23 +1032,21 @@ def _invoke_deployment(cmd, resource_group_name, deployment_name, template, para
     properties = DeploymentProperties(template=template, parameters=parameters, mode='incremental')
     smc = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES,
                                   subscription_id=subscription_id).deployments
+
+    Deployment = cmd.get_models('Deployment', resource_type=ResourceType.MGMT_RESOURCE_RESOURCES)
+    deployment = Deployment(properties=properties)
+
     if validate:
         logger.info('==== BEGIN TEMPLATE ====')
         logger.info(json.dumps(template, indent=2))
         logger.info('==== END TEMPLATE ====')
-
-    if cmd.supported_api_version(min_api='2019-10-01', resource_type=ResourceType.MGMT_RESOURCE_RESOURCES):
-        Deployment = cmd.get_models('Deployment', resource_type=ResourceType.MGMT_RESOURCE_RESOURCES)
-        deployment = Deployment(properties=properties)
-
-        if validate:
-            validation_poller = smc.validate(resource_group_name, deployment_name, deployment)
+        if cmd.supported_api_version(min_api='2019-10-01', resource_type=ResourceType.MGMT_RESOURCE_RESOURCES):
+            validation_poller = smc.begin_validate(resource_group_name, deployment_name, deployment)
             return LongRunningOperation(cmd.cli_ctx)(validation_poller)
-        return sdk_no_wait(no_wait, smc.create_or_update, resource_group_name, deployment_name, deployment)
 
-    if validate:
-        return smc.validate(resource_group_name, deployment_name, properties)
-    return sdk_no_wait(no_wait, smc.create_or_update, resource_group_name, deployment_name, properties)
+        return smc.validate(resource_group_name, deployment_name, deployment)
```

Comment:
For old api version, your new SDK package still use `validate` as function name? Typically with new track2 SDK, all api versions for long running operation should use begin _xxx().

Reply:
Yes, this is because before version `2019-10-01`, `DeploymentsOperations.validate` is a normal synchronous operation. And only after or equal this version, this function is the long running operation.

Swagger definition before version `2019-10-01`: [Swagger link](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resources/resource-manager/Microsoft.Resources/stable/2019-08-01/resources.json#L259)
Swagger definition after (or equal) version `2019-10-01`: [Swagger link](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resources/resource-manager/Microsoft.Resources/stable/2020-10-01/resources.json#L260)